src/components/Molecules/CtaPrompt/CtaPrompt.component.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file CtaPrompt.component.js
+ * Exports a CTA Prompt component.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import CtaMessage from '../CtaMessage/CtaMessage.component';
+
+const cookiePrefix = 'cta_prompt_';
+
+/**
+ * Component that renders a CTA (Call to action) prompt.
+ */
+export default class CtaPrompt extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    type: CtaMessage.propTypes.type
+  };
+
+  static defaultProps = {
+    children: [],
+    type: null
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      shownMessage: null
+    };
+
+    if (props.children) {
+      const children = (props.children.length
+        ? props.children
+        : [props.children]
+      ).filter(child => child.type === CtaMessage);
+
+      // Determine which (if any) message should be shown.
+      children.forEach(msg => {
+        const { data } = msg.props;
+
+        const hash = this.getCookie(data.name);
+
+        // Show first message that either:
+        // - Doesn't have a cookie.
+        // - Cookie value doesn't match message hash.
+        if (!this.state.shownMessage && (!hash || hash !== data.hash)) {
+          this.state.shownMessage = msg;
+        }
+      });
+    }
+  }
+
+  getCookie = name => {
+    const value = `; ${document.cookie}`;
+    const parts = value.split(`; ${cookiePrefix}${name}=`);
+    return parts.length === 2
+      ? parts
+          .pop()
+          .split(';')
+          .shift()
+      : null;
+  };
+
+  setCookie = (name, hash) => {
+    document.cookie = `${cookiePrefix}${name}=${hash}; path=/`;
+  };
+
+  closePrompt = () => {
+    const { shownMessage } = this.state;
+    // Set cookie for message id with value of hash.
+    const { name, hash } = shownMessage.props.data;
+    this.setCookie(name, hash);
+
+    // Close prompt
+    this.setState({ shownMessage: null });
+  };
+
+  render() {
+    const { shownMessage } = this.state;
+    const output = shownMessage
+      ? React.cloneElement(shownMessage, {
```

Comment:
I suspect you might not need to clone shownMessage...

Reply:
After reading through this, I'm thinking maybe you should consider merging CtaMessage and  CtaPrompt. Is there a requirement that the CtaPrompt display multiple messages? Even if/when that is a requirement, you could just render two Cta components and let them handle their own interactions with cookies, etc.

Reply:
There is a requirement to allow multiple messages to be setup for a region, but only show one at a time. Rendering the first message that hasn't been closed by the user (or has changed since the last time user closed the message.) The CtaPrompt acts as coordinator of this logic for the messages.

Reply:
Having them separate allows the CtaMessage component to be used for messages that do not require being tied to a cookie.